File: app/src/main/java/com/mmjang/ankihelper/data/dict/FormsDatabase.java
     package com.mmjang.ankihelper.data.dict;

     import android.content.Context;

     import androidx.room.Database;
     import androidx.room.Room;
     import androidx.room.RoomDatabase;

     @Database(entities = {DummyEntity.class}, version = 1, exportSchema = false)
     public abstract class FormsDatabase extends RoomDatabase {

         public abstract FormsDao formsDao();

         private static FormsDatabase instance;

         public static synchronized FormsDatabase getInstance(Context context) {
             if (instance == null) {
                 instance = Room.databaseBuilder(context.getApplicationContext(),
                         FormsDatabase.class, "forms.db")
                         // Removed .allowMainThreadQueries() - use coroutines for async operations
                         .build();
             }
             return instance;
         }
     }

File: app/src/main/java/com/lmyby/ankihelper/data/dict/FormsDatabase.kt
package com.lmyby.ankihelper.data.dict

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

/**
 * Converted to Kotlin as part of Phase 2 data model migration
 */
@Database(entities = [DummyEntity::class], version = 2, exportSchema = false)
abstract class FormsDatabase : RoomDatabase() {

    abstract fun formsDao(): FormsDao

    companion object {
        @Volatile
        private var instance: FormsDatabase? = null

        @JvmStatic
        fun getInstance(context: Context): FormsDatabase {
            return instance ?: synchronized(this) {
                instance ?: Room.databaseBuilder(
                    context.applicationContext,
                    FormsDatabase::class.java,
                    "forms.db"
                )
                    .createFromAsset("databases/forms.db")
                    .fallbackToDestructiveMigration()
                    // Removed .allowMainThreadQueries() - use coroutines for async operations
                    .build().also { instance = it }
            }
        }
    }
}
